pkg/client/unversioned/request.go

```diff
@@ -427,8 +427,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 		return r
 	}
 	for k, v := range params {
-		for _, vv := range v {
-			r.setParam(k, vv)
+		for _, value := range v {
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
```

Comment:
I see. Perhaps we need to fix the converter or make something into a pointer so we can tell the difference between unspecified and deliberately set to ""? I don't *think* we make a distinction in query parameters, but we do for the label selectors in e.g. service objects-- unset means nothing and set-but-empty means everything. So this may be important to get right, or at least expand the comment?